src/ui/public/query_bar/directive/query_bar.js

```diff
@@ -36,18 +45,47 @@ module.directive('queryBar', function () {
         this.submit();
       };
 
-      this.onTypeaheadSelect = (item) => {
-        this.localQuery.query = item;
-        this.submit();
+      this.suggestionTemplate = suggestionTemplate;
+
+      this.handleKeyDown = (event) => {
+        if (['ArrowLeft', 'ArrowRight', 'Home', 'End'].includes(event.key)) {
+          this.updateSuggestions();
+        }
       };
 
-      this.updateTypeaheadItems = () => {
-        const { persistedLog, localQuery: { query } } = this;
-        this.typeaheadItems = persistedLog.get().filter(recentSearch => {
-          return recentSearch.includes(query) && recentSearch !== query;
+      this.updateSuggestions = () => {
+        $timeout(() => {
```

Comment:
Why does this need to run in a `$timeout`?

Reply:
I was curious about that too. I figured it was because the underlying service wasn't using the angular promise library, but it's also calling `$scope.$apply`, so I wouldn't think this needed to be in a `$timeout` as well.